tests/python_tests/test_vlm_pipeline.py

```diff
@@ -717,255 +831,279 @@ def generate(vlm, requests):
-    def test_prepend_native(self, model_and_tag, conversation_requests):
-        vlm, tag = model_and_tag
 
-        def workaround_inconsistent_inference():
-            answers = generate(vlm, conversation_requests)
+@pytest.mark.precommit
+@model_and_tag_parametrize(TAG_INSERTED_BY_TEMPLATE)
+def test_model_tags_representation(ov_pipe_model: VlmModelInfo, cat_tensor: openvino.Tensor):
+    ov_pipe = ov_pipe_model.pipeline
+    model_id = ov_pipe_model.model_id
+    
+    generation_config = _setup_generation_config(ov_pipe, set_eos_token=False)
+    generation_config = ov_pipe.get_generation_config()
+    generation_config.max_new_tokens = 30
+    ov_pipe.set_generation_config(generation_config)
+    prompt = "Describe"
 
-            vlm.start_chat()
-            native_tag0 = vlm.generate(
-                tag(0) + conversation_requests[0][0], images=conversation_requests[0][1], do_sample=False
-            )
-            assert native_tag0.texts == answers[0].texts
-            assert native_tag0.scores == answers[0].scores
-            native_tags1 = vlm.generate(
-                tag(1) + tag(2) + conversation_requests[1][0], images=conversation_requests[1][1], do_sample=False
+    align_with_optimum_cli = {"padding_side": "left", "truncation_side": "left"}
+    if model_id == "qnguyen3/nanoLLaVA":
+        tokenizer = transformers.AutoTokenizer.from_pretrained(model_id, trust_remote_code=True)
+        messages = [
+            {"role": "user", "content": f'<image>\n{prompt}'}
+        ]
+        templated_prompt = tokenizer.apply_chat_template(
+            messages,
+            tokenize=False,
+            add_generation_prompt=True
+        )
+    else:
+        processor = retry_request(
+            lambda: transformers.AutoProcessor.from_pretrained(
+                model_id,
+                trust_remote_code=True,
+                **align_with_optimum_cli,
             )
-            assert native_tags1.texts == answers[1].texts
-            assert native_tags1.scores == answers[1].scores
-            vlm.finish_chat()
+        )
+        templated_prompt = processor.apply_chat_template(
+            [
+                {
+                    "role": "user",
+                    "content": [
+                        {"type": "image"},
+                        {"type": "text", "text": prompt},
+                    ],
+                }
+            ],
+            add_generation_prompt=True,
+        )
+    def workaround_inconsistent_inference():
+        automatic_tags = ov_pipe.generate(prompt, images=[cat_tensor], do_sample=False)
+        reference_tags = ov_pipe.generate(
+            templated_prompt, images=[cat_tensor], apply_chat_template=False, do_sample=False
+        )
+        assert automatic_tags.texts == reference_tags.texts
+        assert automatic_tags.scores == reference_tags.scores
 
-        retry(workaround_inconsistent_inference)
+    retry(workaround_inconsistent_inference)
```

Comment:
```suggestion
    def workaround_inconsistent_inference():
        __tracebackhide__ = True
        automatic_tags = ov_pipe.generate(prompt, images=[cat_tensor], do_sample=False)
        reference_tags = ov_pipe.generate(
            templated_prompt, images=[cat_tensor], apply_chat_template=False, do_sample=False
        )
        assert automatic_tags.texts == reference_tags.texts
        assert automatic_tags.scores == reference_tags.scores

    retry(workaround_inconsistent_inference)
```
Also should be added to retry.
https://docs.pytest.org/en/6.2.x/example/simple.html#writing-well-integrated-assertion-helpers

Reply:
A little bit out of scope, but added them to all assertion functions which need it. inside the file